### PR TITLE
Harden LocalizableString intrinsically against exceptions

### PR DESCRIPTION
**Pull request by JohnHamby**
Seal public methods of LocalizableString, guaranteeing that all actions go through try/catch blocks that catch exceptions and supply default results.

### CODE REVIEW

**Comment by JohnHamby** [comment]
Forgot to mention:

@mavasani @heejaechang @srivatsn might have particular interest.

File: src/Compilers/Core/Portable/Diagnostic/LocalizableString.cs [hunk 1]
```diff
@@ -41,52 +52,45 @@ string IFormattable.ToString(string ignored, IFormatProvider formatProvider)
...
+        {
+            return Equals((object)other);
+        }
+
+        /// <summary>
+        /// Formats the value of the current instance using the optionally specified format. 
+        /// </summary>
+        protected abstract string GetText(IFormatProvider formatProvider);
+
+        protected abstract int GetHash();
```
Comment (mavasani) [inline src/Compilers/Core/Portable/Diagnostic/LocalizableString.cs]: Add doc comments?

**Comment by mavasani** [comment]
Thanks @JohnHamby , this seems like a very good change.

**Comment by heejaechang** [comment]
:+1: nice!

**Comment by srivatsn** [comment]
@dotnet-bot test this please

**Comment by srivatsn** [comment]
The change looks good to me. Jenkins seems to have failed for some unrelated reason. I've reset it. If it's green then I sign off

**Pull request merged**